tools/headerversions.c

```diff
@@ -0,0 +1,80 @@
+/* Updates the given file if any library versions have changed.  This
+ * is important for systemwide updates, such as sqlite3. */
+#include <ccan/err/err.h>
+#include <ccan/read_write_all/read_write_all.h>
+#include <ccan/tal/grab_file/grab_file.h>
+#include <ccan/tal/str/str.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <gmp.h>
+#include <sqlite3.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <zlib.h>
+
+static const char template[] =
+	"/* Generated file by tools/headerversions, do not edit! */\n"
+	"/* GMP version: %s */\n"
+	"/* SQLITE3 version: %u */\n"
+	"/* ZLIB version: %s */\n"
+	"#include <ccan/err/err.h>\n"
+	"#include <gmp.h>\n"
+	"#include <sqlite3.h>\n"
+	"#include <zlib.h>\n"
+	"\n"
+	"static void check_linked_library_versions(void)\n"
```

Comment:
this is marked `static` which i thought made methods 'private' to the file. how does `lightningd` have access to it?

Reply:
Because lightningd literally `#include`s this file.  The norm would be to make this 'static inline' for a header file; I've fixed that now.